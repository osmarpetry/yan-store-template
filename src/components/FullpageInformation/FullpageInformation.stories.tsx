import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import FullpageInformation, { FullpageInformationProps } from './'

export default {
  title: 'FullpageInformation',
  component: FullpageInformation
} as Meta

const Template: Story<FullpageInformationProps> = (args) => (
  <FullpageInformation {...args} />
)

export const FullpageInformationStandard = Template.bind({})
FullpageInformationStandard.args = {
  title: 'Website in works',
  description: 'Come back later'
}
