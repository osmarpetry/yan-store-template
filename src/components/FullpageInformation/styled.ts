import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #444877;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`
export const Title = styled.h1`
  margin-top: 1.5rem;
  font-size: 4.5rem;
`

export const Description = styled.p`
  font-size: 3rem;
  font-weight: 400;
`

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
