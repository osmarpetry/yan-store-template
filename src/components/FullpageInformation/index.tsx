import * as S from './styled'

export default function FullpageInformation({
  title,
  description,
  img = {
    src: '/img/hero-illustration.svg',
    alt: 'Um desenvolvedor de frente para uma tela com c√≥digo.'
  }
}: {
  title: string
  description: string
  img?: {
    src: string
    alt: string
  }
}) {
  return (
    <S.Wrapper>
      <S.Illustration src={img.src} alt={img.alt} />
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
    </S.Wrapper>
  )
}
