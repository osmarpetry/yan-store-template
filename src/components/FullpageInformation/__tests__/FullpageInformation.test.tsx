import { render } from '@testing-library/react'

import FullpageInformation from '../'

describe('<FullpageInformation.test />', () => {
  it('should render the texts', () => {
    const { container, getByRole, getByText } = render(
      <FullpageInformation
        title="Website in works"
        description="Come back later!"
      />
    )

    expect(
      getByRole('heading', { name: /website in works/i })
    ).toBeInTheDocument()

    expect(getByText('Come back later!')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
