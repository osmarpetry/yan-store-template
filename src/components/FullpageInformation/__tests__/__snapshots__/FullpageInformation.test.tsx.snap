// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullpageInformation.test /> should render the texts 1`] = `
.c0 {
  background-color: #444877;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  margin-top: 1.5rem;
  font-size: 4.5rem;
}

.c3 {
  font-size: 3rem;
  font-weight: 400;
}

.c1 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Um desenvolvedor de frente para uma tela com cÃ³digo."
    class="c1"
    src="/img/hero-illustration.svg"
  />
  <h1
    class="c2"
  >
    Website in works
  </h1>
  <p
    class="c3"
  >
    Come back later!
  </p>
</main>
`
